apps/add_two_ints_service/src/main.c

```diff
@@ -10,55 +10,61 @@
 #define RCSOFTCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printk("Failed status on line %d: %d. Continuing.\n",__LINE__,(int)temp_rc);}}
 
-void main(void)
-{	
-	rcl_init_options_t options = rcl_get_zero_initialized_init_options();
-	RCCHECK(rcl_init_options_init(&options, rcl_get_default_allocator()))
+const char * service_name = "/add_two_ints";
 
-	rcl_context_t context = rcl_get_zero_initialized_context();
-	RCCHECK(rcl_init(0, NULL, &options, &context))
+example_interfaces__srv__AddTwoInts_Request req;
+example_interfaces__srv__AddTwoInts_Response res;
 
-	rcl_node_options_t node_ops = rcl_node_get_default_options();
-	rcl_node_t node = rcl_get_zero_initialized_node();
-	RCCHECK(rcl_node_init(&node, "addtowints_server_rcl", "", &context, &node_ops))
+void service_callback(const void * req, void * res){
+  example_interfaces__srv__AddTwoInts_Request * req_in = (example_interfaces__srv__AddTwoInts_Request *) req;
+  example_interfaces__srv__AddTwoInts_Response * res_in = (example_interfaces__srv__AddTwoInts_Response *) res;
 
-	const char * service_name = "add_two_ints";
-	rcl_service_options_t service_op = rcl_service_get_default_options();
-	rcl_service_t serv = rcl_get_zero_initialized_service();
-	const rosidl_service_type_support_t * service_type_support = ROSIDL_GET_SRV_TYPE_SUPPORT(example_interfaces, srv, AddTwoInts);
+  printf("Service request value: %d + %d.\n", (int) req_in->a, (int) req_in->b);
 
-	RCCHECK(rcl_service_init(&serv,&node, service_type_support, service_name, &service_op))
+  res_in->sum = req_in->a + req_in->b;
+}
 
-	rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
-	RCCHECK(rcl_wait_set_init(&wait_set, 0, 0, 0, 0, 1, 0, &context, rcl_get_default_allocator()))
+void main(void)
+{	
+    allocator = rcl_get_default_allocator();
 
-	do {
-		RCSOFTCHECK(rcl_wait_set_clear(&wait_set))
+    // create init_options
+    RCCHECK(rclc_support_init(&support, 0, NULL, &allocator));
 
-		size_t index;
-		RCSOFTCHECK(rcl_wait_set_add_service(&wait_set, &serv, &index))
+    // create node
+    RCCHECK(rclc_node_init_default(&node, "add_twoints_client_rclc", "", &support));
 
-		rcl_wait(&wait_set, RCL_MS_TO_NS(100));
+    // create service
+    RCCHECK(rclc_service_init_default(&service, &node, ROSIDL_GET_SRV_TYPE_SUPPORT(example_interfaces, srv, AddTwoInts), "/addtwoints"));
 
-		if (wait_set.services[index]) {   
-			rmw_request_id_t req_id;
-			example_interfaces__srv__AddTwoInts_Request req;
-			example_interfaces__srv__AddTwoInts_Request__init(&req);
-			RCSOFTCHECK(rcl_take_request(&serv,&req_id,&req))
+    // create executor
+    RCCHECK(rclc_executor_init(&executor, &support.context, 1, &allocator));
 
-			example_interfaces__srv__AddTwoInts_Response res;
-			example_interfaces__srv__AddTwoInts_Response__init(&res);
+    unsigned int rcl_wait_timeout = 10;   // in ms
+    RCCHECK(rclc_executor_set_timeout(&executor, RCL_MS_TO_NS(rcl_wait_timeout)));
```

Comment:
not necessary to set timeout. The default timeout is 1s.

Reply:
One question, if I provide an executor timeout greater than the timeout value set in `rclc_executor_spin_some` which one will be the spin_some timeout?

Reply:
If you do not set any timeout then the default timeout is used (1s). If you use the function rclc_executor_set_timeout(value) then the timeout is overwritten by 'value' - irrespective if it is larger or smaller.

Reply:
When it is called from executor_spin() then the default value (1s) or the value defined by the user is used. If you call the method directly, then you set the timeout: 
See for example here: https://github.com/micro-ROS/rclc/blob/ad5da630ab4900e2b997e02be1ca0d81bebf70c1/rclc/src/rclc/executor.c#L1000

Reply:
Thanks, I understand it now. I'm going to remove those lines.